exec/api/src/main/java/org/jboss/windup/exec/rulefilters/PhaseRulesFilter.java

```diff
@@ -0,0 +1,47 @@
+package org.jboss.windup.exec.rulefilters;
+
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import org.jboss.windup.config.RuleProvider;
+import org.jboss.windup.config.phase.ReportGenerationPhase;
+import org.jboss.windup.config.phase.ReportRenderingPhase;
+import org.jboss.windup.config.phase.RulePhase;
+
+/**
+ * Filters the rules with given phases.
+ *
+ * @author Ondrej Zizka, ozizka at redhat.com
+ */
+public class PhaseRulesFilter implements RuleProviderFilter
```

Comment:
I have added that to javadoc. Making the filter aware of what is the computed phase to execute the rule in would need change in the filter API. Or we would have to put that information upfront into the metadata.